### DISCUSSION TITLE
When we run multiple socket processes how to disconnect a user

### DISCUSSION THREAD

**Discussion by arun6894** [miguelgrinberg/python-socketio#1228] [Q&A] [answered]
When we run multiple socket processes if a user enters a room with 1st process and the same user tries to open a new connection in a new tab with the new process I need to disconnect the 1st process connection.

If 1st process enters a room,  2nd process while disconnecting the 1st process can't leave a room as he doesn't know the sid details. How can we achieve him to leave a room from second process.

As we are storing sid data across every process separately. Is there a way to disconnect during multiple process running.

@miguelgrinberg

**Comment by miguelgrinberg** [comment]
Are you using a message queue? This is required when running multiple processes. With a correctly set up cluster, you can call `disconnect(sid)` on any node, and if the referenced `sid` is in another the nodes communicate to do the disconnection.

**Reply by arun6894** [reply]
I am using Kafka as a message queue. and I am triggering disconnect(sid) during the new connection if the already connection exists stored sid from Redis as well.
This means an existing connection in one process, When trying to get a new connection from another process I try to disconnect the first process sid At the start of the disconnect(sid), I need some data get_session(sid) that is stored in session in order to send it to Kafka.  I get "Session not Found error ". Because of this error i can't leave the room of the first connection. In the end both the connections are alive.

@miguelgrinberg

**Reply by miguelgrinberg** [reply, marked as answer]
Unfortunately the user sessions are not currently available across processes, but a disconnection request should work just fine regardless of the process it is issued from. You may need to implement a different storage for your user session data if you need to access it from other processes.

**Reply by arun6894** [reply]
Okay, Understood.
Below are methods used to store session and get session data.  Don't we think it is storing the data to a local variable which leads to an increase in memory even though we clear the data during disconnections What if there are huge numbers of connections in idle and new connections keep increasing the memory and it gradually increases in RAM leading to memory leakage? Any thoughts on this.
**self.save_session(sid, session)
self.get_session(sid)**

**Reply by miguelgrinberg** [reply]
The user sessions are stored in memory and it is provided as a convenience for applications that want to associate data with each connection, such as the user's ID and/or roles. If you want to store a large amount of data per client then this is not an appropriate place. In that case you should store your data in a database.
